### ISSUE TITLE
[Feature] Implement a DNS crawler feature

### ISSUE THREAD

**Issue by dangershony** [stratisproject/StratisBitcoinFullNode#649]
This [crawler](https://github.com/dev0tion/stratis-seeder) is  a network crawler that will keep track of active nodes on the network and server them to A record requests.

We could create a feature (as nuget package) that can be attached to nodes that act as seed nodes and keep track of active nodes (outbound nodes) and server the list nodes on requests.

Found this library in C# to create a light [DNS server](https://github.com/kapetan/dns) 
Note NBitcoin has support for DNS messages as well

**Comment by mtaipe** [comment]
This is the reason why other peers didn't connect to my node. Noticed that after running stratisd for a while and then started FN again, incoming connections appeared. Seems that stratisd has an API that let the seeder learn about them but the FN doesn't have such API yet.

**Comment by D3m0nKingx** [comment]
I can further add to this, that when I was initializing the FN with `-connect=ip` argument that sometimes there was only one (outbound) connection that I specified and no other connections. Other times it would have the IP I specified (outbound) and a lot of inbound connections.   

Afaik there isn't any code implemented that temporarily bans peers based on connection issues? (going off and on intermittently)
